### PR TITLE
Connecting redux

### PR DESCRIPTION
**Pull request by glebdronchenko**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+import Geolocation from './services/GeolocationService';
+import ApiService from './services/ForecastApiService';
+import { changeLocation, setGeolocationCity } from './actions/locationsManagerActions';
+import Storage from './services/StorageConnectionService';
+import { CURRENT_LOCATION_STORAGE_CODE } from './utils/constants';
+
+function App(props) {
+  useEffect(async () => {
+    props.setGeolocationCity();
+  }, []);
```
Comment (anton-vlasik) [inline src/App.jsx]: you'll need to have setGeolocationCity listed in the dependency array. Take a look here https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

File: src/components/CurrentCityForecast/CurrentCityForecast.jsx [hunk 2]
```diff
@@ -12,23 +12,31 @@ import {
...
 } from '../../utils/constants';
 
-function CurrentCityForecast() {
+function CurrentCityForecast({ currentLocation }) {
   const [currentCityForecast, setCurrentCityForecast] = useState({});
 
-  const currentLocationId = mockLocation.id;
-  const symbolUrl = `${FORECAST_SYMBOL_LINK}${currentCityForecast.forecast?.symbol}${FORECAST_SYMBOL_EXT}`;
+  const currentLocationId = currentLocation.id;
+  console.log(currentLocation);
```
Comment (anton-vlasik) [inline src/components/CurrentCityForecast/CurrentCityForecast.jsx]: remove console.log

File: src/components/CurrentCityForecast/components/HourlyForecastChart/HourlyForecastChart.jsx [hunk 3]
```diff
@@ -9,9 +9,15 @@ function HourlyForecastChart({ locationId }) {
   const [hours, temperatures] = prepareChartData(hourlyForecast);
 
   useEffect(async () => {
-    const { data } = await ApiService.getHourlyForecast(locationId);
-    setHourlyForecast(data.forecast);
-  }, []);
+    try {
+      if (locationId) {
+        const { data } = await ApiService.getHourlyForecast(locationId);
```
Comment (anton-vlasik) [inline src/components/CurrentCityForecast/components/HourlyForecastChart/HourlyForecastChart.jsx]: Again, start moving this outside. Not sure if you were going to do that in this PR or in another

File: src/components/Header/components/LocationSearch/components/SearchDropDown/SearchDropDown.jsx [hunk 4]
```diff
@@ -16,35 +15,17 @@ import { COOKIE_TOKEN_FIELD } from '../../../../../../utils/constants';
...
   const [searchString, setSearchString] = useState('');
   const [locations, setLocations] = useState([]);
-  const [cookies] = useCookies([COOKIE_TOKEN_FIELD]);
 
   useEffect(() => {
     const setLocationsValue = async (): Promise<void> => {
-      const searchedResult: SearchedLocationsType = await getSearchLocations(searchString, cookies);
-      setLocations(searchedResult.locations);
+      const { data } = await ApiService.getLocationsSearch(searchString);
+      console.log(data);
```
Comment (anton-vlasik) [inline src/components/Header/components/LocationSearch/components/SearchDropDown/SearchDropDown.jsx]: remove console.logs

File: src/components/Warnings/Warnings.jsx [hunk 5]
```diff
@@ -20,12 +20,13 @@ function Warnings({ currentLocation }: WarningsPropsType): React$Node {
   useEffect(() => {
     const getWarningsData = async (): Promise<void> => {
       try {
+        console.log('currentLocation.id: ', currentLocation.id);
```
Comment (anton-vlasik) [inline src/components/Warnings/Warnings.jsx]: remove console.logs

File: src/reducers/locationManagerReducers.js [hunk 6]
```diff
@@ -1,8 +1,15 @@
...
+import StorageConnection from '../services/StorageConnectionService';
+import { CURRENT_LOCATION_STORAGE_CODE, FAVORITE_CITIES_STORAGE_CODE } from '../utils/constants';
+
+const currentLocationFromStore = StorageConnection.getValue(CURRENT_LOCATION_STORAGE_CODE);
+const favoriteCitiesFromStore = StorageConnection.getValue(FAVORITE_CITIES_STORAGE_CODE);
 
 const initialState = {
-  currentLocation: {},
-  searchString: ''
+  currentLocation: currentLocationFromStore ? currentLocationFromStore : {},
```
Comment (anton-vlasik) [inline src/reducers/locationManagerReducers.js]: I don't suggest to use conditions in your initial state

File: src/reducers/locationManagerReducers.js [hunk 7]
```diff
@@ -1,8 +1,15 @@
 import { CHANGE_LOCATION, CHANGE_SEARCH_STRING } from '../actions/locationsManagerActions';
+import StorageConnection from '../services/StorageConnectionService';
+import { CURRENT_LOCATION_STORAGE_CODE, FAVORITE_CITIES_STORAGE_CODE } from '../utils/constants';
+
+const currentLocationFromStore = StorageConnection.getValue(CURRENT_LOCATION_STORAGE_CODE);
+const favoriteCitiesFromStore = StorageConnection.getValue(FAVORITE_CITIES_STORAGE_CODE);
```
Comment (anton-vlasik) [inline src/reducers/locationManagerReducers.js]: This operation may fail. What will be your initial state set to then?

File: src/App.jsx (hunk 1 above)
Comment (glebdronchenko) [reply src/App.jsx]: Done.

File: src/components/CurrentCityForecast/CurrentCityForecast.jsx (hunk 2 above)
Comment (glebdronchenko) [reply src/components/CurrentCityForecast/CurrentCityForecast.jsx]: Done.

File: src/components/Header/components/LocationSearch/components/SearchDropDown/SearchDropDown.jsx (hunk 4 above)
Comment (glebdronchenko) [reply src/components/Header/components/LocationSearch/components/SearchDropDown/SearchDropDown.jsx]: Done.

File: src/components/CurrentCityForecast/components/HourlyForecastChart/HourlyForecastChart.jsx (hunk 3 above)
Comment (glebdronchenko) [reply src/components/CurrentCityForecast/components/HourlyForecastChart/HourlyForecastChart.jsx]: Done. I have moved it to actions.

File: src/components/Warnings/Warnings.jsx (hunk 5 above)
Comment (glebdronchenko) [reply src/components/Warnings/Warnings.jsx]: Done.

File: src/reducers/locationManagerReducers.js (hunk 6 above)
Comment (glebdronchenko) [reply src/reducers/locationManagerReducers.js]: Done. I have moved this conditionals to expressions where I am getting data from local storage.

File: src/reducers/locationManagerReducers.js (hunk 7 above)
Comment (glebdronchenko) [reply src/reducers/locationManagerReducers.js]: I have moved conditionals from initial state to these operations.

**Review by anton-vlasik — APPROVED** [review]

**Review by ximet — APPROVED** [review]

**Pull request merged**